src/DicomMetaDictionary.js

```diff
@@ -149,7 +149,8 @@ class DicomMetaDictionary {
                     if (
                         sqZero &&
                         typeof sqZero === "object" &&
-                        !sqZero.length
+                        !sqZero.length &&
+                        !(sqZero instanceof ArrayBuffer)
```

Comment:
@wayfarer3130 can you look and test?

Reply:
That makes sense for Seg, or anything else that decodes direct to raw buffers, still would like to see a unit test.

Reply:
addAccessors function is fine, the problem comes from the conditions to which it's being called. Before, it would make `naturalDataset[naturalName] = naturalDataset[naturalName][0] `and now it leaves it as is which will breaks downstream code and make it undefined.

Have no ideia why it made sense before to remove the element out of the array but code was made with that assumption and the proposed safeguard just tries to make it backward compatible.

anyway, I'll put a test later if it helps